Add message directing users to security guide

arcalinea opened the pull request:
Addresses #2142, which was blocking on updates to zcash/support/security.html. That page has now been added, so this message directing users to the site can be included. 

It displays in the zcash-cli --help and --version message text, and on the zcashd metrics screen.

arcalinea commented:
Output of `zcash-cli --help`
![help-securitymsg](https://cloud.githubusercontent.com/assets/12245008/24276728/f5ebd730-0ff5-11e7-9282-6bd625178b7e.png)

Output of `zcash-cli --version`
![version-security-msg](https://cloud.githubusercontent.com/assets/12245008/24276734/ff5bbe0c-0ff5-11e7-8535-26e76a577830.png)

arcalinea commented:
Metrics console:

![metrics-securitymsg](https://cloud.githubusercontent.com/assets/12245008/24276782/4b9cb794-0ff6-11e7-9427-6a9daac8ab06.png)

daira commented on src/metrics.cpp:
```diff
@@ -330,6 +330,11 @@ void ThreadShowMetricsScreen()
+        // Security warning text
+        std::cout << std::endl;
+        std::cout << _("In order to ensure you are adequately protecting your privacy when using Zcash,") << std::endl;
+        std::cout << _("please see https://z.cash/support/security.html") << std::endl;
```
Since it's a single sentence, this should probably be a single translation string.

daira replied:
https://z.cash/support/security/index.html (in case the redirect ever goes away).

daira commented on src/bitcoin-cli.cpp:
```diff
@@ -65,6 +65,8 @@ static bool AppInitRPC(int argc, char* argv[])
     if (argc<2 || mapArgs.count("-?") || mapArgs.count("-h") || mapArgs.count("-help") || mapArgs.count("-version")) {
         std::string strUsage = _("Zcash RPC client version") + " " + FormatFullVersion() + "\n";
+        strUsage += "\n" + _("In order to ensure you are adequately protecting your privacy when using Zcash,");
+        strUsage += _("please see https://z.cash/support/security.html") + "\n";
```
Same comment as below.

arcalinea commented:
@daira made the changes requested and rebased

str4d commented on src/bitcoin-cli.cpp:
```diff
@@ -65,6 +65,7 @@ static bool AppInitRPC(int argc, char* argv[])
     ParseParameters(argc, argv);
     if (argc<2 || mapArgs.count("-?") || mapArgs.count("-h") || mapArgs.count("-help") || mapArgs.count("-version")) {
         std::string strUsage = _("Zcash RPC client version") + " " + FormatFullVersion() + "\n";
+        strUsage += "\n" + _("In order to ensure you are adequately protecting your privacy when using Zcash, please see https://z.cash/support/security/index.html") + "\n";
```
`strprintf(_("..., please see %s"), "https://z.cash/support/security/index.html")` so the URL isn't part of the translation string.

str4d replied:
Also needs another newline between this and the copyright notice, which should just be added at the top of the copyright notice so there isn't an additional space for `--help`.

str4d commented on src/metrics.cpp:
```diff
@@ -330,6 +330,10 @@ void ThreadShowMetricsScreen()
+
+        // Security warning text
+        std::cout << std::endl;
+        std::cout << _("In order to ensure you are adequately protecting your privacy when using Zcash, please see https://z.cash/support/security/index.html") << std::endl;
```
Ditto.

arcalinea commented:
Addressed @str4d 's comments. 
- put the security message in the actual license text itself with a line separating it from the license info (and it's still displayed in `zcash-cli help`)
- removed the url from the translation string

daira commented on src/bitcoin-cli.cpp:
```diff
@@ -66,6 +66,8 @@ static bool AppInitRPC(int argc, char* argv[])
     if (argc<2 || mapArgs.count("-?") || mapArgs.count("-h") || mapArgs.count("-help") || mapArgs.count("-version")) {
         std::string strUsage = _("Zcash RPC client version") + " " + FormatFullVersion() + "\n";
         if (!mapArgs.count("-version")) {
+            strUsage += "\n" + strprintf(_("In order to ensure you are adequately protecting your privacy when using Zcash, please see %s"), "https://z.cash/support/security/index.html") + "\n";
```
Add call to FormatParagraph here.

daira commented on src/metrics.cpp:
```diff
@@ -330,6 +330,10 @@ void ThreadShowMetricsScreen()
+
+        // Security warning text
+        std::cout << std::endl;
+        std::cout << strprintf(_("In order to ensure you are adequately protecting your privacy when using Zcash, please see %s"), "https://z.cash/support/security/index.html") << std::endl;
```
Add call to FormatParagraph here.

daira commented:
Review needed for 5a0e8bb154ba8a14c6fe4bed34fbde9795f25301 .

str4d commented on src/bitcoin-cli.cpp:
```diff
@@ -64,18 +64,14 @@ static bool AppInitRPC(int argc, char* argv[])
     ParseParameters(argc, argv);
     if (argc<2 || mapArgs.count("-?") || mapArgs.count("-h") || mapArgs.count("-help") || mapArgs.count("-version")) {
-        std::string strUsage = _("Zcash RPC client version") + " " + FormatFullVersion() + "\n";
+        std::string strUsage = _("Zcash RPC client version") + " " + FormatFullVersion() + "\n" + LicenseInfo();
```
Why include this here?

- It's a sizeable inclusion, particularly for `zcash-cli`.
- It almost completely removes the need for `zcash-cli --version`.
- I'm pretty sure this will break `help2man`, or at least cause issues for it during the release process.

daira commented:
@str4d addressed your comment (``-help`` no longer shows license text for either zcashd or zcashd-cli, but ``zcashd -help`` does show the privacy notice). Force-pushed, please re-review.

daira commented:
@zkbot r+
